src/basic_pro.cpp

```diff
@@ -2107,43 +2107,46 @@ static DWORD launch_cmd(BOOL hide, BOOL nowait,
         e->Throw("UNIT kw. relies on GNU extensions to the std C++ library (that were not available during compilation?)");
 #endif
       } else {
-        FILE *coutF=NULL, *cerrF=NULL;
+        FILE *coutF=nullptr, *cerrF=nullptr;
         if (nParam > 1) {
           coutF = fdopen(coutP[0], "r");
-          if (coutF == NULL) close(coutP[0]);
+          if (coutF == nullptr) close(coutP[0]);
         }
         if (nParam > 2) {
           cerrF = fdopen(cerrP[0], "r");
-          if (cerrF == NULL) close(cerrP[0]);
+          if (cerrF == nullptr) close(cerrP[0]);
         }
 
         vector<DString> outStr;
         vector<DString> errStr;
 
         // read cout
-        if (nParam > 1 && coutF != NULL) {
-          while (fgets(buf, bufSize, coutF) != NULL) {
+        if (nParam > 1 && coutF != nullptr) {
+          while (fgets(buf, bufSize, coutF) != nullptr) {
             SizeT len = strlen(buf);
             if (len != 0 && buf[len - 1] == '\n') buf[len - 1] = 0;
-            outStr.push_back(DString(buf));
+            outStr.emplace_back(buf);
           }
           fclose(coutF);
         }
 
         // read cerr
-        if (nParam > 2 && cerrF != NULL) {
-          while (fgets(buf, bufSize, cerrF) != NULL) {
+        if (nParam > 2 && cerrF != nullptr) {
+          while (fgets(buf, bufSize, cerrF) != nullptr) {
             SizeT len = strlen(buf);
             if (len != 0 && buf[len - 1] == '\n') buf[len - 1] = 0;
-            errStr.push_back(DString(buf));
+            errStr.emplace_back(buf);
           }
           fclose(cerrF);
         }
 
         // wait until child terminates
-        int status;
-        pid_t wpid = wait(&status);
+        int status = -1;
+        if (waitpid(pid, &status, 0) == -1) {
+          e->Throw(DString("Error waiting for child process: ") + strerror(errno));
+        }
 
+        // if waitpid fails, we still have status in the default of -1, indicating a failure
```

Comment:
This is from an earlier version with a Warning issued instead of an exception being thrown.
```suggestion
```